Source/Evolve/WarpXEvolve.cpp

```diff
@@ -548,14 +548,14 @@ WarpX::OneStep_multiJ (const amrex::Real cur_time)
     if (WarpX::fft_do_time_averaging) PSATDEraseAverageFields();
 
     // 3) Deposit rho (in rho_new, since it will be moved during the loop)
-    if (WarpX::update_with_rho)
+    if (update_with_rho && rho_in_time == RhoInTime::Linear)
```

Comment:
As discussed offline: we would probably check the pointer on `rho` here instead.

Reply:
Yes, will have to rebase after #3544.